openedx/core/djangoapps/credentials/tasks/v1/tasks.py

```diff
@@ -54,3 +62,178 @@ def send_grade_to_credentials(self, username, course_run_key, verified, letter_g
             f"Failed to send grade to credentials. Reason: {error_msg}"
         )
         raise self.retry(exc=exception, countdown=countdown, max_retries=MAX_RETRIES)
+
+
+@shared_task(bind=True, ignore_result=True)
+@set_code_owner_attribute
+def handle_notify_credentials(self, options, course_keys):
+    """
+    Celery task to handle the notify_credentials management command. Finds the
+    relevant cert and grade records, then starts other celery tasks to send the
+    data.
+    """
+
+    try:
+        site_config = SiteConfiguration.objects.get(site__domain=options['site']) if options['site'] else None
+    except SiteConfiguration.DoesNotExist:
+        logger.error('No site configuration found for site %s', options['site'])
```

Comment:
❓ I noticed that this is fragile. (This is unchanged from the previous implementation.) If I pass the site flag with garbage (`--site garbledigoop`), the program logs the error but then continues along, eventually failing completely on line 104 below because `site_config` is not defined at all.

Seems like we should either:
- set `site_config = None` here so that the program can continue, or
- fail more intentionally at this point. (`logger.fatal` instead of `.error`? Consider how an invalid course key is [handled in this command here](https://github.com/edx/edx-platform/blob/3660b2a34619a0f23ddc28578eef08f2e69dafde/openedx/core/djangoapps/credentials/management/commands/notify_credentials.py#L343-L345).)

Reply:
`return`ed instead of `raise`d because the logs ended up being unnecessarily redundant. Here’s what this error will look like:

```
root@lms:/edx/app/edxapp/edx-platform# ./manage.py lms notify_credentials --user_ids 3 --notify_programs --verbose --site garbagenonexistantsite

# ... removed some irrelevant warnings ...

2021-04-13 20:11:04,098 INFO 83090 [openedx.core.djangoapps.credentials.management.commands.notify_credentials] [user None] [ip None] notify_credentials.py:152 - notify_credentials starting, dry-run=False, site=garbagenonexistantsite, delay=0 seconds, page_size=100, from=NA, to=NA, notify_programs=True, user_ids=['3'], execution=manual
2021-04-13 20:11:04,098 INFO 83090 [openedx.core.djangoapps.credentials.management.commands.notify_credentials] [user None] [ip None] notify_credentials.py:187 - 0 courses specified:
2021-04-13 20:11:04,139 ERROR 83090 [openedx.core.djangoapps.credentials.tasks.v1.tasks] [user None] [ip None] tasks.py:79 - No site configuration found for site garbagenonexistantsite
Traceback (most recent call last):
  File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/credentials/tasks/v1/tasks.py", line 77, in handle_notify_credentials
    site_config = SiteConfiguration.objects.get(site__domain=options['site']) if options['site'] else None
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/db/models/query.py", line 406, in get
    raise self.model.DoesNotExist(
openedx.core.djangoapps.site_configuration.models.SiteConfiguration.DoesNotExist: SiteConfiguration matching query does not exist.
2021-04-13 20:11:04,141 INFO 83090 [celery.app.trace] [user None] [ip None] trace.py:125 - Task openedx.core.djangoapps.credentials.tasks.v1.tasks.handle_notify_credentials[ee197a34-8412-4556-8b19-4cc18380ee53] succeeded in 0.02421850000973791s: None

Reply:
Should we add a default to the edx site? If not we need to update the auto-config for this job, I do not think it provides a site.

Even without the site, I believe the script would work fine with a Course or a user. It might try to send extra data though.